### PR TITLE
chore(deps): upgrade to node 16

### PR DESCRIPTION
**Pull request by kennylam**
### Related Ticket(s)

#9059 

### Description

This raises the requirement for Node to version 16.

### Changelog

**Changed**

- Node requirement bumped to `v16`
- `node-sass` bumped to `6.0.1`
- various dependency bumps to patch versions

### CODE REVIEW

File: .github/workflows/deploy-canary.yml [hunk 1]
```diff
@@ -74,10 +74,15 @@ jobs:
       - uses: actions/checkout@v3
         with:
           token: ${{secrets.MERGE_ACTION}}
+<<<<<<< HEAD
       - name: Use Node.js 14.x
         uses: actions/setup-node@v3
+=======
+      - name: Use Node.js 16.x
+        uses: actions/setup-node@v2
+>>>>>>> a9064d28cd (chore(deps): upgrade to node 16)
```
Comment (abdonrd) [inline .github/workflows/deploy-canary.yml:83]: Be careful with this :)

(there is more like this in the PR)

Comment (kennylam) [reply .github/workflows/deploy-canary.yml:83]: Huh, thanks for pointing this out! Not sure what's up, my editor didn't show any conflicts.

File: .github/workflows/deploy-next.yml [hunk 2]
```diff
@@ -74,10 +74,15 @@ jobs:
       - uses: actions/checkout@v3
         with:
           token: ${{secrets.MERGE_ACTION}}
+<<<<<<< HEAD
       - name: Use Node.js 14.x
         uses: actions/setup-node@v3
+=======
+      - name: Use Node.js 16.x
+        uses: actions/setup-node@v2
+>>>>>>> a9064d28cd (chore(deps): upgrade to node 16)
```
Comment (abdonrd) [inline .github/workflows/deploy-next.yml:83]: One more!

File: .github/workflows/deploy-next.yml [hunk 3]
```diff
@@ -150,10 +155,15 @@ jobs:
...
```
Comment (abdonrd) [inline .github/workflows/deploy-next.yml:164]: One more!

File: .github/workflows/e2e-integration.yml [hunk 4]
```diff
@@ -45,11 +45,17 @@ jobs:
...
     steps:
+<<<<<<< HEAD
       - uses: actions/checkout@v3
       - name: Use Node.js 14.x
         uses: actions/setup-node@v3
+=======
+      - uses: actions/checkout@v2
+      - name: Use Node.js 16.x
+        uses: actions/setup-node@v2
+>>>>>>> a9064d28cd (chore(deps): upgrade to node 16)
```
Comment (abdonrd) [inline .github/workflows/e2e-integration.yml:58]: One more!

**Review by abdonrd — CHANGES_REQUESTED** [review]

**Comment by kennylam** [comment]
Closing in favor of https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/10239.

**Pull request closed without merging**